keboola/component_configuration_resource.go

```diff
@@ -0,0 +1,426 @@
+package keboola
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/keboola/go-client/pkg/client"
+	. "github.com/keboola/go-client/pkg/storageapi"
+	"github.com/keboola/go-utils/pkg/orderedmap"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = &configResource{}
+	_ resource.ResourceWithConfigure = &configResource{}
+	//_ resource.ResourceWithImportState = &configResource{}
+)
+
+// NewConfigResource is a helper function to simplify the provider implementation.
+func NewConfigResource() resource.Resource {
+	return &configResource{}
+}
+
+// configResource is the resource implementation.
+type configResource struct {
+	sapiClient *client.Client
+}
+
+// Config https://keboola.docs.apiary.io/#reference/components-and-configurations/component-configurations/list-configurations
+type configModel struct {
+	ID                types.String `tfsdk:"id"`
+	BranchID          types.Int64  `tfsdk:"branch_id"`
+	ComponentID       types.String `tfsdk:"component_id"`
+	ConfigID          types.String `tfsdk:"configuration_id"`
+	Name              types.String `tfsdk:"name"`
+	Description       types.String `tfsdk:"description"`
+	ChangeDescription types.String `tfsdk:"change_description"`
+	IsDeleted         types.Bool   `tfsdk:"is_deleted"`
+	Created           types.String `tfsdk:"created"`
+	Version           types.Int64  `tfsdk:"version"`
+	//State             types.Map    `tfsdk:"state" readonly:"true"`
+	IsDisabled types.Bool   `tfsdk:"is_disabled"`
+	Content    types.String `tfsdk:"configuration"`
+}
+
+func getConfigModelId(model *configModel) string {
+	return fmt.Sprintf("%d/%s/%s", model.BranchID, model.ComponentID, model.ID)
+}
+
+// Metadata returns the resource type name.
+func (r *configResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_component_configuration"
+}
+
+// GetSchema defines the schema for the resource.
+func (r *configResource) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Description: "Manages component configurations (https://keboola.docs.apiary.io/#reference/components-and-configurations).",
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Description: "Unique string identifier assembled as branchId/componentId/configId.",
+				Type:        types.StringType,
+				Computed:    true,
+				PlanModifiers: []tfsdk.AttributePlanModifier{
+					resource.UseStateForUnknown(),
+				},
```

Comment:
`id` je viacmenej povinny atribut terraform resourcu. V tomto pripade je to konkatenacia branchId, componentId a configId.